.travis.yml

```diff
@@ -6,48 +6,40 @@ cache:
 
 os: linux
 
-node_js:
-  # We test the latest version on circleci
-  - "11"
-  - "10"
-  - "8"
-  - "6"
-
 env:
   global:
     - PATH=$HOME/.yarn/bin:$PATH
     - JOB=test
 
-before_install:
-  - curl -o- -L https://yarnpkg.com/install.sh | bash
-
-install:
-  # the `make test-ci` script runs this command already
-  - if [ "$JOB" != "test" ] && [ "$JOB" != "lint" ]; then yarn install; fi
-  - if [ "$JOB" = "lint" ]; then make bootstrap; fi
-
+install: skip
 before_script:
+  - curl -o- -L https://yarnpkg.com/install.sh | bash
   - if [ "$TRAVIS_OS_NAME" = "windows" ]; then choco install make; fi
-  - if [ "$JOB" = "babel-parser-flow-tests" ]; then make bootstrap-flow; fi
-  - if [ "$JOB" = "babel-parser-test262-tests" ]; then make bootstrap-test262; fi
 
 script:
-  - if [ "$JOB" = "test" ]; then make test-ci; fi
-  - if [ "$JOB" = "lint" ]; then make lint && make flow; fi
-  - if [ "$JOB" = "babel-parser-flow-tests" ]; then make test-flow-ci; fi
-  - if [ "$JOB" = "babel-parser-test262-tests" ]; then make test-test262-ci; fi
+  - if [ "$JOB" = "test" ]; then make -j test-ci; fi
+  - if [ "$JOB" = "lint" ]; then make -j code-quality-ci; fi
+  - if [ "$JOB" = "babel-parser-flow-tests" ]; then make -j test-flow-ci; fi
+  - if [ "$JOB" = "babel-parser-test262-tests" ]; then make -j test-test262-ci; fi
 
 matrix:
   fast_finish: true
   include:
+    - node_js: "node"
+      env: JOB=lint
+  # We test the latest version on circleci
+    - node_js: "11"
+    - node_js: "10"
+  # Move `windows` build to be the fourth since it is slow
```

Comment:
Also node 6 is slow. What do you think about this order?

- Lint
- Node 11 (fast, for quick feedback)
- Windows
- Node 6
- Node 8
- Node 10
- Flow
- Test262

Reply:
We can definitely give it a try. As long as `Windows` or `Node 6` is not the last I think they would not make significant difference.

How many concurrent jobs do we have on circle ci? If we have two concurrent jobs we can move windows testing to circle ci.

Reply:
4 I think